ipa-core/src/protocol/context/batcher.rs

```diff
@@ -1,50 +1,43 @@
-use std::{
-    collections::VecDeque,
-    sync::atomic::{AtomicUsize, Ordering},
-};
+use std::{cmp::min, collections::VecDeque, future::Future};
 
-use tokio::sync::Notify;
+use bitvec::{bitvec, prelude::BitVec};
+use tokio::sync::watch;
 
 use crate::{
+    error::Error,
     protocol::RecordId,
     sync::{Arc, Mutex},
 };
 
-pub enum Either<L, R> {
-    Left(L),
-    Right(R),
-}
-
-impl<L, R> Either<L, R> {
-    fn left(value: L) -> Self {
-        Self::Left(value)
-    }
-
-    fn right(value: R) -> Self {
-        Self::Right(value)
-    }
-}
-
 #[derive(Debug)]
-pub struct BatchState<B> {
+pub(super) struct BatchState<B> {
```

Comment:
It would be good to add some docs on these structs. What they do and how they coordinate together.